test/test_docker_policy_run_args.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+NORM='\033[0m'
+
+# pull the image now so that we don't have to guess/query when its done in the script
+docker pull osrf/ros:dashing-desktop
+
+# run the example; loads listener in Docker
+./examples/mem_limit_sandbox_docker.launch.py&
+task_id=$!
+
+echo "TaskID: $task_id"
+
+echo "Checking if sandboxed-listener-node is running..."
+docker inspect -f "{{.State.Running}}" sandboxed-listener-node
+is_running=$?
+
+# Wait until the docker container is running. docker inspect will return 0 when it does.
+while [[ $is_running -ne 0 ]]
+do
+  sleep 1
+  docker inspect -f "{{.State.Running}}" sandboxed-listener-node
+  is_running=$?
+done
+
+# Run listener in the container spun up by DockerPolicy for 5 seconds
+echo "Checking memory limits in Docker container..."
+memory=$(docker inspect sandboxed-listener-node | jq '.[0].HostConfig.Memory')
+if [[ "$memory" -eq "134217728" ]]; then
+    printf "${GREEN}Memory limits correctly set to 128m!${NORM}\n";
+else
+    printf "${RED}Memory limits not correctly set to 128m!${NORM}\n";
+fi
+
+echo "Stopping Docker container..."
+
+# Note: these sleep commands are here just so the following command executes after stdout appears on
+# the terminal.
+kill -INT $task_id
+sleep 2
+
+# SIGTERM is required also if launch is ran in the background, this might be a bug in launch.
+kill $task_id
+sleep 2
+
+echo "Checking if sandboxed-listener-node is running..."
+docker inspect -f "{{.State.Running}}" sandboxed-listener-node
+
+# Check the exit code of docker inspect. 0 is returned only if it is running.
+# This check will set the exit code to 0 only if the exit code is not 0.
+[[ $? -ne 0 ]]
+
+exit $?
```

Comment:
The exit code should be determined if the check on line 31 passed.
lines 53, 55 return success if line 49 fails.

Reply:
This was my thought as well.
Should there be a function that shuts down the Docker container (since we don't want to leave a zombie container) which we call in the memory checker block and then return based on the result of the test?